tests/php/RedirectFileControllerTest.php

```diff
@@ -0,0 +1,453 @@
+<?php
+
+namespace SilverStripe\Assets\Tests;
+
+use SilverStripe\Assets\Image;
+use SilverStripe\Assets\Storage\AssetStore;
+use SilverStripe\Assets\Storage\ProtectedFileController;
+use SilverStripe\Assets\Folder;
+use SilverStripe\Assets\Filesystem;
+use SilverStripe\Assets\File;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\Dev\FunctionalTest;
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Assets\Tests\Storage\AssetStoreTest\TestAssetStore;
+
+/**
+ * @skipUpgrade
+ */
+class RedirectFileControllerTest extends FunctionalTest
+{
+    protected static $fixture_file = 'FileTest.yml';
+
+    protected $autoFollowRedirection = false;
+
+    public function setUp()
+    {
+        parent::setUp();
+
+        // Set backend root to /ImageTest
+        TestAssetStore::activate('RedirectFileControllerTest');
+
+        // Create a test folders for each of the fixture references
+        foreach (Folder::get() as $folder) {
+            /** @var Folder $folder */
+            $filePath = TestAssetStore::getLocalPath($folder);
+            Filesystem::makeFolder($filePath);
+        }
+
+        // Create a test files for each of the fixture references
+        foreach (File::get()->exclude('ClassName', Folder::class) as $file) {
+            /** @var File $file */
+            $path = TestAssetStore::getLocalPath($file);
+            Filesystem::makeFolder(dirname($path));
+            $fh = fopen($path, "w+");
+            fwrite($fh, str_repeat('x', 1000000));
+            fclose($fh);
+
+            // Create variant for each file
+            $this->getAssetStore()->setFromString(
+                str_repeat('y', 100),
+                $file->Filename,
+                $file->Hash,
+                'variant'
+            );
+        }
+    }
+
+    public function tearDown()
+    {
+        TestAssetStore::reset();
+        parent::tearDown();
+    }
+
+    public function testLegacyFilenameRedirect()
+    {
+        $file = File::find('FileTest-subfolder/FileTestSubfolder.txt');
+
+        $response = $this->get('/assets/FileTest-subfolder/FileTestSubfolder.txt');
+        $this->assertResponse(
+            404,
+            '',
+            false,
+            $response,
+            'Legacy URL for unpublished file should return 404'
+        );
+
+        $file->publishSingle();
+        $response = $this->get('/assets/FileTest-subfolder/FileTestSubfolder.txt');
+        $this->assertResponse(
+            307,
+            '',
+            $file->getURL(false),
+            $response,
+            'Legacy URL for published file should return 307'
+        );
+
+        $response = $this->get($response->getHeader('location'));
+        $this->assertResponse(
+            200,
+            str_repeat('x', 1000000),
```

Comment:
This is a bit noisy - could it just be `version 1` as content? Or would that mess with the test setup?

Reply:
That's copied over from some other test. I can change it to `version 1`.

Reply:
Yeah you're building more complexity on that baseline, so I think it'd be appropriate to change

Reply:
Change to `version 1`